FWCore/FWCore/DataWrapper.h

```diff
@@ -23,8 +25,7 @@ class GAUDI_API DataWrapper : public DataWrapperBase {
   const T* getData() {return m_data;} 
   void setData(T* data) {m_data = data;}
 
-  virtual albers::CollectionBase* collectionBase(){return dynamic_cast<albers::CollectionBase*>(m_data); }
-
+  virtual albers::CollectionBase* collectionBase(){if (std::is_base_of<albers::CollectionBase,T>::value) return reinterpret_cast<albers::CollectionBase*>(m_data); return nullptr;}
```

Comment:
As a second thought: There should maybe be some documentation that this may return a nullptr.